backend/src/main/resources/db/migrations/V1__remove_old_email_index.sql

```diff
@@ -0,0 +1 @@
+alter table administrators drop constraint administrators_email_unique;
```

Comment:
It might be different for this specific setup, but in all migration frameworks I know the initial migration starts with an empty database and creates all tables and columns to allow a deterministic outcome (or is this done as part of the `setupDatabase*` functions?). Otherwise, this might even throw an error on a new setup where the `administrators_email_unique` constraint doesn't exist yet?

Of course this would normally cause an error on all systems which already have initialized the database, so they have to be [baselined](https://documentation.red-gate.com/fd/baseline-184127456.html) (or the new initial migration would have to be applied with the [skipExecutingMigrations](https://documentation.red-gate.com/fd/skip-executing-migrations-184127537.html)) option to tell flyway that it should mark the initial migration as applied without really changing anything.
This file would then be v2 and could be applied on top of the initial one.

Reply:
Jep you're completely right. It must be at least V2 to be properly applied. Tested that again today. The setupDatabase* functions will create all the tables. Correct. But as this is coupled into the kotlin code and nondeterministic, meaning there can be a different outcome when applying the setupDatabase function e.g. initially and on an already existing Database, As the tables are created in kotlin directly, and columns and constraints are added automatically when we add them in java. See [jetbrains exposed](https://github.com/JetBrains/Exposed).